wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WFileWidget.java

```diff
@@ -79,6 +84,15 @@ public void setFileTypes(final String[] types) {
 	public void setFileTypes(final List<String> types) {
 		getOrCreateComponentModel().fileTypes = types;
 	}
+	
+	/**
+	 * @see setFileTypes(List<>)
+	 * @see setFileTypes(String[])
+	 * @return {@code} true if one or more file type is supplied.
+	 */
+	public boolean hasFileTypes() {
```

Comment:
A small performance improvement is to call `getComponentModel` once as it can be expensive.